torchvision/csrc/io/decoder/stream.cpp

```diff
@@ -49,7 +51,51 @@ int Stream::openCodec(std::vector<DecoderMetadata>* metadata) {
                << ", avcodec_alloc_context3 failed";
     return AVERROR(ENOMEM);
   }
+  // multithreading heuristics
+  // this is a dumb conservative limit; ideally we'd use 
+  // int max_threads = at::get_num_threads(); but this would cause 
+  // fb sync to fail as it would add dependency to ATen to the decoder API
+  int max_threads = 12;
```

Comment:
Should we do a `#define` for this instead of using a magic number inside a function?

Reply:
I suppose we can yes; is it customary to do that in the header file?

Reply:
Do you suspect max_threads will be used elsewhere? Perhaps declaring it a `const` might be better(could be under namespace ffmpeg.

Reply:
No, not really. At least not in the same context.

Reply:
I've defined `max_threads` as the protected const int in the header file until we find a better way of solving this.